### PR TITLE
New park form

### PR DESCRIPTION
**Pull request by snardella**
- Added 422 validation for backend
- Added a post route for /api/v1/parks/ to handle new park form post
- Added localhoust:3000/new to client router.  Changed from parks/new to /new since parks/new was confusing the router.
- Changed default length of our picture field from 256 to 1000000 to hold long image url addresses at the database level.
- Changed address for “Add New Park!” top bar to goto /new.
- Added image to ParkTile.js and removed the url from the tile.
- Changed AuthenticatedRoute from /parks/new to /new

### CODE REVIEW

File: client/src/components/App.js [hunk 1]
```diff
@@ -41,6 +42,7 @@ const App = (props) => {
         <Route exact path="/users/new" component={RegistrationForm} />
         <Route exact path="/user-sessions/new" component={SignInForm} />
         <AuthenticatedRoute exact path="/profile" component={UserProfile} user={currentUser} />
+        <AuthenticatedRoute exact path="/new" component={NewParkForm} user={currentUser} />
```
Comment (bkincart) [inline client/src/components/App.js]: Hey y'all! I would actually argue that this path SHOULD be `/parks/new` in order to stick to RESTful conventions. Can you tell me a little more about why `/parks/new` was giving you errors?

File: client/src/components/NewParkForm.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`;
+          const error = new Error(errorMessage);
+          throw error;
+        }
+      } else {
+        const body = await response.json();
+        console.log("posted successfully", body)
+        setShouldRedirect(true)
+
```
Comment (bkincart) [inline client/src/components/NewParkForm.js]: Remove empty line

File: client/src/components/NewParkForm.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        console.log("posted successfully", body)
+        setShouldRedirect(true)
+       
+      }
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`);
+    }
+  };
+
+
```
Comment (bkincart) [inline client/src/components/NewParkForm.js]: Please clean up whitespace here: have only one empty line here, and one in between `handleInputChange` and `handleSubmit`

File: client/src/components/NewParkForm.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    setNewPark({
+      name: "",
+      description: "",
+      location: "",
+      rating: "",
+      picture: "",
+    })
+  }
+
+
```
Comment (bkincart) [inline client/src/components/NewParkForm.js]: Remove extra empty line

File: client/src/components/UserProfile.js [hunk 5]
```diff
@@ -8,6 +8,8 @@ const UserProfile = (props) => {
   return (
     <div>
       <h1>Parkview profile for {userName}</h1>
+      <Link to="/parks/new">Add a new Park!</Link>
```
Comment (bkincart) [inline client/src/components/UserProfile.js:11]: This is linking to `/parks/new`, which I actually think is great, but it IS mismatching your current route :)

File: server/src/routes/api/v1/parksRouter.js [hunk 6]
```diff
@@ -12,6 +15,20 @@ parksRouter.get("/", async (req, res) => {
   }
 })
 
+parksRouter.post("/", async (req, res) => {
+  const { body } = req
+  const formInput = cleanUserInput(body)
+  try{
```
Comment (bkincart) [inline server/src/routes/api/v1/parksRouter.js]: Please clean up whitespace here: spaces between `try {` and `} catch` for example

**Review by bkincart — CHANGES_REQUESTED** [review]

File: client/src/components/App.js (hunk 1 above)
Comment (snardella) [reply client/src/components/App.js]: So for some reason when we posted to /parks/new it was also hitting the parksRouter.get("/:id"... and was sending back a 500 error saying it couldn't fetch a park with id "new".  As I'm typing it I'm sure there's a reason why it was hitting the get request and that should probably be figured out...

Comment (bkincart) [reply client/src/components/App.js]: Ahhhh yes 😄  this is a tricky one. Dynamic routes can often cause this issue. Order matters here!

If you actually move this route (line 45) ABOVE the dynamic route (line 41), that should fix it for ya :)

**Pull request merged**